### PR TITLE
[IMP] mail: Rename temporary to uploading in attachments related content

### PR DESCRIPTION
**Pull request by jgi-odoo**
This commit renames "temporary" into "uploading" in attachments content

task-2337268

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jgi-odoo** [comment]
@alexkuhn @seb-odoo

**Review by seb-odoo — COMMENTED** [review]
Almost perfect, I would just also rename `getAttachmentNextTemporaryId` and its internal variable.

Note this function is not ideal and should be refactored in another task. Maybe if we need an id, we should actually request a new id from the server before uploading its corresponding data.

File: addons/mail/static/src/components/attachment/attachment.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
                 'o-downloadable': props.isDownloadable,
                 'o-editable': props.isEditable,
                 'o-has-card-details': detailsMode === 'card',
-                'o-temporary': attachment and attachment.isTemporary,
+                'o-uploading': attachment and attachment.isUploading,
```
Comment (alexkuhn) [inline addons/mail/static/src/components/attachment/attachment.xml:10]: `o-isUploading` (it matches with naming convention I'm introducing with my WIP PR)

File: addons/mail/static/src/components/composer/composer_tests.js [hunk 2]
```diff
@@ -1936,7 +1936,7 @@ QUnit.test('remove an uploading attachment', async function (assert) {
     );
     assert.containsOnce(
         document.body,
-        '.o_Composer .o_Attachment.o-temporary',
+        '.o_Composer .o_Attachment.o-uploading',
```
Comment (alexkuhn) [inline addons/mail/static/src/components/composer/composer_tests.js:1939]: Since diff in this LOC, can remove the `.o_Composer` part, which is useless

**Review by alexkuhn — CHANGES_REQUESTED** [review]

File: addons/mail/static/src/components/attachment/attachment.xml (hunk 1 above)
Comment (jgi-odoo) [reply addons/mail/static/src/components/attachment/attachment.xml]: Done

File: addons/mail/static/src/components/composer/composer_tests.js (hunk 2 above)
Comment (jgi-odoo) [reply addons/mail/static/src/components/composer/composer_tests.js]: Done

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**